hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -434,6 +515,29 @@ private ListKeysResponse listKeys(ListKeysRequest request, int clientVersion)
     return resp.build();
   }
 
+  @RequestFeatureValidator(
+      conditions = ValidationCondition.OLDER_CLIENT_REQUESTS,
+      processingPhase = RequestProcessingPhase.POST_PROCESS,
+      requestType = Type.ListKeys
+  )
+  public static OMResponse disallowListKeysResponseWithECReplicationConfig(
```

Comment:
this check is here because of the fears in my prev answer for the disallowInfoBucketResponseWithECReplicationConfig method, and the reasoning in the answer below this one by two for disallowListBucketsResponseWithECReplicationConfig.

Reply:
KeyArgs and BucketArgs involve different proto changes. In BucketArgs, one new field is added to the proto. In KeyArgs a new field is added, but a new value was also added to the ReplicationType enum in hdds.proto, which KeyArgs stores as the field called `type`. For this reason, I think an old client would be able to handle BucketArgs with EC info, but not KeyArgs with EC info. We should test the two cases separately to make sure.

Reply:
Test I believe made this clear and bucket listing and info is not disabled anymore